### PR TITLE
implemented pkce, axios post issue resolved

### PR DESCRIPTION
**Pull request by amans330**
state was not required in token request, causing axios to fail. resolved now.

### CODE REVIEW

File: server/helpers/pkce.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const crypto = require('crypto');
+
```
Comment (mooreds) [inline server/helpers/pkce.js:2]: Hmm. Seems like we use this a couple of different places. Is there a pkce npm package? Should we add one if there's not?

File: server/routes/login.js [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 const router = express.Router();
+const pkce = require('../helpers/pkce');
 
 router.get('/', (req, res) => {
+  // Generate and store the PKCE verifier
+  req.session.verifier = pkce.generateVerifier();
+  // Generate the PKCE challenge
+  const challenge = pkce.generateChallenge(req.session.verifier);
 
   const stateValue = Math.random().toString(36).substring(2,15) + Math.random().toString(36).substring(2,15) + Math.random().toString(36).substring(2,15) + Math.random().toString(36).substring(2,15) + Math.random().toString(36).substring(2,15) + Math.random().toString(36).substring(2,15);
```
Comment (mooreds) [inline server/routes/login.js:11]: This would be good to extract to a lib as well.

File: server/routes/oauth-callback.js [hunk 3]
```diff
@@ -1,9 +1,46 @@
-
 const express = require("express");
 const router = express.Router();
+const request = require('request');
```
Comment (mooreds) [inline server/routes/oauth-callback.js:3]: Why do we need this lib? I looked and didn't see it used.

File: server/routes/oauth-callback.js [hunk 4]
```diff
@@ -1,9 +1,46 @@
...
 const axios = require("axios").default;
 const qs = require("query-string");
 
+// router.get('/', (req, res) => {
```
Comment (mooreds) [inline server/routes/oauth-callback.js:7]: If we comment out all of this, we should just remove it.

File: server/routes/oauth-callback.js [hunk 5]
```diff
@@ -12,39 +49,36 @@ const config = {
...
     .then((result) => {
-
       // save token to session
       req.session.token = result.data.access_token;
-
       //redirect to Vue app
      res.redirect(`http://localhost:8081`);
     })
     .catch((err) => {
+        // console.log("aman")
```
Comment (mooreds) [inline server/routes/oauth-callback.js:81]: Nit, please clean up debug statements :)

**Review by mooreds — CHANGES_REQUESTED** [review]
Please update the readme with info about us using PKCE as well.

**Comment by amans330** [comment]
There is a pkce npm package. Looking into it. Working on the rest of the changes as well.

**Comment by amans330** [comment]
Hi @mooreds made the changes, please have a look.

**Review by mooreds — APPROVED** [review]

**Pull request merged**